haskell.nix/flake.nix

```diff
@@ -15,102 +15,127 @@
+
+        ## nixpkgs has weeder 2, but we use weeder 1
+        #weeder-legacy = pkgs.haskellPackages.callHackageDirect {
+        #  pkg = "weeder";
+        #  ver = "1.0.9";
+        #  sha256 = "0gfvhw7n8g2274k74g8gnv1y19alr1yig618capiyaix6i9wnmpa";
+        #} {};
+
+        ## a derivation which generates a script for running weeder
+        #weeder-script = weeder-hacks.weeder-script {
+        #  weeder = weeder-legacy;
+        #  hs-pkgs = hs-pkgs;
+        #  local-packages = [
+        #    { inherit name; subdirectory = "."; }
+        #  ];
+        #};
+
+      in {
+
+        ## In case the package contains exes,
+        # packages = hs-pkg.components.exes;
+
+        ## In case the package contains some "main" exe that shares the name with the package,
+        # defaultPackage = hs-pkg.components.exes.${name};
+
+        # nixpkgs revision pinned by this flake
+        legacyPackages = pkgs;
+
+        # derivations that we can run from CI, or locally with `nix flake check`
+        checks = {
+          # builds all haskell components
+          build-all = pkgs.buildEnv {
+            name = "${name}-all-components";
+            ignoreCollisions = true;
+            paths = all-components;
           };
 
-        }];
-      };
-
-      hs-pkg = hs-pkgs.pataq-package;
-
-      # returns the list of all components for a package
-      get-package-components = pkg:
-        # library
-        lib.optional (pkg ? library) pkg.library
-        # haddock
-        ++ lib.optional (pkg ? library) pkg.library.haddock
-        # exes, tests and benchmarks
-        ++ lib.attrValues pkg.exes
-        ++ lib.attrValues pkg.tests
-        ++ lib.attrValues pkg.benchmarks;
-
-      # all components for the current haskell package
-      all-components = get-package-components hs-pkg.components;
-
-      # for weeder:
-      ##weeder-hacks = import haskell-nix-weeder { inherit pkgs; };
-
-      # nixpkgs has weeder 2, but we use weeder 1
-      ##weeder-legacy = pkgs.haskellPackages.callHackageDirect {
-      ##  pkg = "weeder";
-      ##  ver = "1.0.9";
-      ##  sha256 = "0gfvhw7n8g2274k74g8gnv1y19alr1yig618capiyaix6i9wnmpa";
-      ##} {};
-
-      # a derivation which generates a script for running weeder
-      ##weeder-script = weeder-hacks.weeder-script {
-      ##  weeder = weeder-legacy;
-      ##  hs-pkgs = hs-pkgs;
-      ##  local-packages = [
-      ##    { name = "pataq-package"; subdirectory = "."; }
-      ##  ];
-      ##};
-
-    in {
-      # nixpkgs revision pinned by this flake
-      legacyPackages.x86_64-linux = pkgs;
-
-      # derivations that we can run from CI
-      checks.x86_64-linux = {
-        # builds all haskell components
-        build-all = all-components;
-
-        # runs the test
-        test = hs-pkg.checks.pataq-test;
-      };
-
-      # script for running weeder
-      ##weeder-script = weeder-script;
-    };
+          # runs the tests
+        } // hs-checks;
```

Comment:
Should we `// self.packages.${system}` here as well, to build everything with `nix flake check` in a single step?

Reply:
There's already `checks.${system}.build-all`